deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts

```diff
@@ -130,7 +131,13 @@ export class MigrationServiceCore extends Stack {
             portMappings: props.portMappings,
             logging: LogDrivers.awsLogs({
                 streamPrefix: `${props.serviceName}-logs`,
-                logGroup: serviceLogGroup
+                logGroup: serviceLogGroup,
+                // E.g. "[INFO ] 2024-12-31 23:59:59,99999999"
+                // and  "[ERROR] 2024-12-31 23:59:59,999"
+                // and  "2024-12-31 23:59:59"
+                multilinePattern: "^(\\[[A-Z ]{1,5}\\] )?\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}(,\\d{2,8})?",
+                // Log appenders are non-blocking so defer async behavior to them
```

Comment:
This comments describes the opposite of what is on the AWS docs - https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_awslogs.html

Maybe remove the comment?

I guess the question is - is it OK to have the container be unable to continue executing while logs are committed to the cloudwatch log system?  It seems like the answer is YES - the logs are critical to the operation of the container and we'd rather no more work take place while the logs are committed to storage.

Reply:
see https://github.com/opensearch-project/opensearch-migrations/pull/602#discussion_r1576770648

Reply:
My mental model here is that spikes in logging latency should not result in spikes in application latency (definitely for proxy, maybe debatable for replay).

That contributed to the change for an async base logger. How we behave when that buffer fills up and isn't drained can be a different story

Reply:
I still think the comment is incorrect, what do you think about removing it?

Reply:
What about "Defer buffering behavior to log4j2 for greater flexibility".

Blocking is the default behavior so I could remove that definition all together, but I was wanting to be explicit if we would rely on buffering/dropping from another part of the system